solutions/ac-GCDOnBlackboard.mdx

```diff
@@ -0,0 +1,63 @@
+---
+id: ac-GCDOnBlackboard
+source: AC
+title: GCD On Blackboard
+author: Andrew Cheng
+---
+
+**Time Complexity:**
+$\mathcal{O}(N+\log(max \ a_i))$
```

Comment:
`\max` everywhere

Reply:
can you explain why this is the time complexity :)

Reply:
My understanding is that the GCD operation reaches the maximum time complexity of ```O(log \max a_i)``` when GCD(a,b) = 1 (consecutive numbers in the fibonacci sequence if I recall correctly). Throughout each case in the brute force algorithm, the lowest possible GCD of the remaining ```N-1``` numbers is 1. Since it takes ```O(log \max a_i)``` operations to get the GCD to 1, the total time complexity of each case using brute force should be ```O(N + log(\max a_i)``` instead of ```O(Nlog(\max a_i))```. Is this correct ?

Reply:
basically yes, this explanation is more complete though: https://codeforces.com/blog/entry/92720?#comment-814416